R/caretStack.R

```diff
@@ -55,80 +54,95 @@ caretStack <- function(all.models, ...){
 #' \dontrun{
 #' library("rpart")
 #' models <- caretList(
-#'   x=iris[1:100,1:2],
-#'   y=iris[1:100,3],
-#'   trControl=trainControl(method="cv"),
-#'   methodList=c("rpart", "glm")
+#'   x = iris[1:100, 1:2],
+#'   y = iris[1:100, 3],
+#'   trControl = trainControl(method = "cv"),
+#'   methodList = c("rpart", "glm")
 #' )
-#' meta_model <- caretStack(models, method="lm")
-#' RMSE(predict(meta_model, iris[101:150,1:2]), iris[101:150,3])
+#' meta_model <- caretStack(models, method = "lm")
+#' RMSE(predict(meta_model, iris[101:150, 1:2]), iris[101:150, 3])
 #' }
 predict.caretStack <- function(
-  object, newdata=NULL,
-  se=FALSE, level=0.95,
-  return_weights=FALSE,
-  na.action=na.omit,
-  ...){
+    object, newdata = NULL,
+    se = FALSE, level = 0.95,
+    return_weights = FALSE,
+    na.action = na.omit,
+    ...) {
   stopifnot(is(object$models, "caretList"))
   type <- extractModelTypes(object$models)
 
-  #preds <- do.call(predict, c(list(object=object$models, newdata=newdata), modelList_predict_params)) TODO keep this around for future use
-  preds <- predict(object$models, newdata=newdata, na.action=na.action)
-  if(type == "Classification"){
-    out <- predict(object$ens_model, newdata=preds, na.action=na.action, ...)
-    # Need a check here
-    if(class(out) %in% c("data.frame", "matrix")){
-      # Return probability predictions for only one of the classes
-      # as determined by configured default response class level
-      est <- out[, getBinaryTargetLevel(), drop = TRUE]
-    } else{
-      est <- out
+  # preds <- do.call(predict, c(list(object=object$models, newdata=newdata), modelList_predict_params)) TODO keep this around for future use
+  preds <- predict(object$models, newdata = newdata, na.action = na.action)
+
+  if (type == "Classification") {
+    column_names <- colnames(preds)
+    num_classes <- length(levels(object$models[[1]]$pred$obs))
+    if (getMulticlassExcludedLevel() >= 1 && getMulticlassExcludedLevel() <= num_classes) {
```

Comment:
Can you add some comments explaining this logic?

Reply:
Done